Docs updates -- Do not merge (#149)
Azure/azure-saas: docs/azure-saas-docs/content/en/components/admin-service.md

  - Depends on the SaaS.Permissions.Service for CRUD operations on permissions records
- [Identity Provider](../identity/identity-provider)
  - Depends on the identity provider to authenticate callers via their JWT token

### Consumers

Currently, the only consumer of this API is the 2 frontend applications. Every module in the ASDK project was designed to be extensible, so you could also build your own applications that interface directly with this API to fit other use cases that the included frontend applications do not provide.

- [SaaS.SignupAdministration.Web](../signup-administration)

- [Saas.Application.Web](../saas-application)

### Authentication

The Admin Service is secured using OAuth 2.0 authentication via the Microsoft Identity Platform. Incoming requests must contain a valid JWT Bearer token on the `Authorization` header. The token must contain a valid scope that the calling application has been authorized to use. To learn more about how this process works and is configured in B2C, we highly reccomend checking out our list of [identity resources & documentation]((../../resources/additional-recommended-resources#identity-focused)).

> For authorization, we have also included middleware on the Admin Service that extracts the user's permission records from the JWT token claims and preforms authorization based on policies applied at the route level. In other words, before preforming any action on a tenant once a request is received, the admin service will first ensure that the user making the request has a claim to that tenant on their token. If there is no claim to that tenant, or their role does not match what they're trying to do, the request will be denied.

Reviewer: ```suggestion
For authorization, we have also included middleware on the Admin Service that extracts the user's permission records from the JWT token claims and preforms authorization based on policies applied at the route level. In other words, before performing any action on a tenant as requested by a user, the admin service will first ensure that the user making the request has a claim to that tenant on their token. If there is no claim to that tenant or their role does not match what they're trying to do, the request will be denied.
```

In addition to Arsen's suggestion.

> Implementing this in a multitenant fashion is often the most difficult part of starting a SaaS project, so we tried to make it as extensible as possible to support a wide range of scenarios.

Reviewer: This being one of the most difficult aspects, it would greatly benefit users of the kit to include some further reading on particular scenarios under Additional Reading at the bottom of the page.

Author: In the paragraph above this one, it should link to the additional identity reading
